user/user_lego_sensor.h

```diff
@@ -16,9 +16,18 @@
 #include <lego.h>
 #include <lego_sensor_class.h>
 
+/* The USER_LEGO_SENSOR_TEXT_VALUE_SIZE can be a maximum of PAGE_SIZE-2
+ * to take into account a trailing newline (added later) and NULL. We are
+ * limiting it to much less to save memory for each user_lego_sensor 
+ * instance.
+ */
+
+#define USER_LEGO_SENSOR_TEXT_VALUE_SIZE (512-2)
```

Comment:
I'm going to leave it at 512 bytes for now - there's still an issue with memory management after the file is unlinked. Once that's sorted out I'll fine tune the string allocation